### PR TITLE
feat: integrate next layout

### PR DESCRIPTION
**Pull request by ruimonteiro93**
Integration of next-layout in the boilerplate, allowing for the use of customizable layouts in the projects built from it.
The following additions were performed:

- A default layout with a simple header and footer
- Updated documentation to what’s included

### CODE REVIEW

File: docusaurus/docs/what-is-included/internationalization.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ We have chosen [`react-intl`](https://github.com/formatjs/react-intl/) for inter
 
 The integration of `react-intl` into Next.js is made with [@moxy/next-intl](https://github.com/moxystudio/next-intl).
 
+ℹ️ If you are really sure internationalization is not needed in your project, you'll want to remove all the unnecessary `intl` related code. Start by uninstalling `react-intl` and `@moxy/next-intl`. You can then proceed to do a global search for both `react-intl` and `@moxy/next-intl` to find all their occurrences across the project, and clean the places where they are not being used anymore.
```
Comment (satazor) [inline docusaurus/docs/what-is-included/internationalization.md]: I think we should mention that we should remove the root `intl` folder. Also, we no longer need the `full-icu` package and it’s usage in the npm scripts in the package.json.

File: docusaurus/docs/what-is-included/internationalization.md [hunk 2]
```diff
@@ -27,3 +29,4 @@ Policies control which locale is active at a given point, amongst other things.
 You may want to create custom policies for certain use-cases. One common use-case is to have a policy that matches against the locale saved in the account preferences of authenticated users.
 
 Please check `@moxy/next-intl`'s [documentation](https://github.com/moxystudio/next-intl#custom-policies) to know more.
+
```
Comment (satazor) [inline docusaurus/docs/what-is-included/internationalization.md]: Please remove this extra line.

File: docusaurus/docs/what-is-included/customizable-layout-support.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+id: customizable-layout-support
+title: Customizable Layout Support 
+sidebar_label: Customizable Layout Support
+---
+
+The boilerplate includes customizable layout support with [@moxy/next-layout](https://github.com/moxystudio/next-layout/). This easily allows for your application to have a layout applied to all pages. If you don't want to have all pages with the same layout though, you can develop another one as multiple layouts are also supported. Furthermore, `@moxy/next-layout` also supports nested layouts, so if you want to put a layout inside a layout, you're perfectly fine to do so.
```
Comment (satazor) [inline docusaurus/docs/what-is-included/customizable-layout-support.md]: Missing quotes in the link.

File: docusaurus/docs/what-is-included/customizable-layout-support.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (satazor) [inline docusaurus/docs/what-is-included/customizable-layout-support.md]:
```suggestion
The boilerplate includes customizable layout support with [`@moxy/next-layout`](https://github.com/moxystudio/next-layout/). This easily allows for your application to have a layout applied to all pages. If you don't want to have all pages with the same layout though, you can develop another one as multiple layouts are also supported. Furthermore, `@moxy/next-layout` also supports nested layouts, so if you want to put a layout inside a layout, you're perfectly fine to do so.
```

File: docusaurus/docs/what-is-included/customizable-layout-support.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+title: Customizable Layout Support 
+sidebar_label: Customizable Layout Support
+---
+
+The boilerplate includes customizable layout support with [`@moxy/next-layout`](https://github.com/moxystudio/next-layout/). This easily allows for your application to have a layout applied to all pages. If you don't want to have all pages with the same layout though, you can develop another one as multiple layouts are also supported. Furthermore, `@moxy/next-layout` also supports nested layouts, so if you want to put a layout inside a layout, you're perfectly fine to do so.
+
+A `MainLayout` component, with a simple header and footer, comes with the boilerplate and is the default layout. You can modify it in whatever way you see fit. Let's say you want your footer to be optional, so that you can hide it in some pages. Just add a `withFooter` prop to `MainLayout`, and then wrap the page in which you want to hide the footer with `withLayout`:
+
+```js
+export default withLayout(<MainLayout withFooter={ false } />)(About)
```
Comment (satazor) [inline docusaurus/docs/what-is-included/customizable-layout-support.md]:
```suggestion
export default withLayout(<MainLayout withFooter={ false } />)(About);
```

File: docusaurus/docs/what-is-included/internationalization.md [hunk 6]
```diff
@@ -27,3 +27,15 @@ Policies control which locale is active at a given point, amongst other things.
...
+## Removing this feature
+
+If you are really sure internationalization is not needed in your project, you'll want to remove all the unnecessary `intl` related code. Be sure to follow these steps in order to clean your project properly:
+
+1. Uninstall `react-intl`, `@moxy/next-intl` and `full-icu`.
+2. Remove the `NODE_ICU_DATA` environment variable from both `dev` and `start` scripts, found in `package.json`.
+3. Remove the `intl` folder.
+4. Search globally for `react-intl` and `@moxy/next-intl` and remove the corresponding code across the project
```
Comment (satazor) [inline docusaurus/docs/what-is-included/internationalization.md]:
```suggestion
4. Search globally for `react-intl` and `@moxy/next-intl` and remove the corresponding code across the project.
```

**Comment by jgradim** [comment]
@ruimonteiro93 We still need to add an entry to `docusaurus/sidebars.js` for `what-is-included/customizable-layout-support.md`, otherwise this page will not show up when browsing the documentation.

Other than that, everything seems ok!

**Pull request merged**